lib/resources.js

```diff
@@ -554,6 +564,46 @@ const buildElasticBody = function (params) {
   return body
 }
 
+/**
+ * Given params which include an identifier number, returns query for that identifier
+ */
+
+const queryForIdentifierNumber = function (params) {
+  if (params.lccn) {
+    return {
+      query: {
+        regexp: {
+          idLccn: {
+            value: `[^\\d]*${regexEscape(params.lccn)}[^\\d]*`
+          }
+        }
+      }
+    }
+  } else if (params.issn) {
+    return { query: { bool: { must: { term: { idIssn: params.issn } } } } }
+  } else if (params.isbn) {
+    return { query: { bool: { must: { term: { idIsbn: params.isbn } } } } }
+  } else if (params.oclc) {
+    return {
+      query: {
+        regexp: {
+          'identifierV2.value': {
+            value: `.*\\(OCoLC\\)${regexEscape(params.oclc)}([^\\d].*)?`
+          }
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Given a string, escapes all regex control characters
+ */
+
+const regexEscape = function (str) {
+  return str.replace(/[-\/\\^$*+?.()|[\]{}]/g, (match) => { return '\\' + match })
```

Comment:
This feels more readable, but curious if there's a functional difference with using the previous `.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&')`

Reply:
There should be no difference, I just find it more readable.